compiler/crates/relay-compiler/src/build_project/artifact_writer.rs

```diff
@@ -232,3 +233,76 @@ impl ArtifactWriter for NoopArtifactWriter {
         Ok(())
     }
 }
+
+pub struct ArtifactValidationWriter {
+    added: Mutex<Vec<PathBuf>>,
+    updated: Mutex<Vec<PathBuf>>,
+    removed: Mutex<Vec<PathBuf>>,
```

Comment:
What do you think about changing this to DashSet? https://docs.rs/dashmap/latest/dashmap/struct.DashSet.html

I think it will be easier to work with (no explicit `unwrap`, etc).

Reply:
Yeah, great idea!